### ISSUE TITLE
Modulo with Double-values fails (result = 0)

### ISSUE THREAD

**Issue by GWRon** [bmx-ng/bcc#113]
NG fails to do "doubleVar Mod value" (result is always "0")

```BlitzMax
SuperStrict
Framework Brl.StandardIO
Import Brl.Math

Local timeGone:Double = 2058048061.0000
'this would result in 13 days but results in "0"
Print Int((Floor((timeGone / 86400) * 24)) Mod 24)
'because "mod" fails for doubles (should print "13.000...00")
Print timeGone Mod 24
```

For "float" and "long" it works as expected.

thought "fmod" (the one @woollybah  uses for "mod" in c) handles "doubles" already so the problem might be something different:

Generated C-code (extract):
```C
		BBDOUBLE bbt_timeGone=2058048061.0000000f;
		brl_standardio_Print(bbStringFromInt((BBINT)fmod(bbFloor((bbt_timeGone/86400.000000000000f)*24.000000000000000f), 24.000000000000000f)));
		brl_standardio_Print(bbStringFromDouble(fmod(bbt_timeGone, 24.000000000000000f)));

```

**Comment by woollybah** [comment]
Yes. The generated code looks as I'd expect.
Running the same code on legacy on Windows, I get 0 and 0 as answers too.

Note that when converting the value 2058048061.0000 to a Long, it becomes, 2058048000

This may be where your issues are coming from?

**Comment by GWRon** [comment]
Just to give "why" I post these bugs -and that I often assume that it works in vanilla:

I hope you recognized that I do try to get my game running: and it has for now some flaws with NG. As these flaws are not appearing with "vanilla" (for 99% of my users and my test computers) I assume that it does not happen with the constellation of my compiler-chain.

But of course you are right: I edited the sample to make it sure that it is a double (the ".0000" part) but in my tests I had it without ".0000" so the value was without decimals:

```BlitzMax
SuperStrict
Framework Brl.StandardIO
Import Brl.Math

Local timeGone:Double = 2058048061
'this would result in 13 days but results in "0"
Print Int((Floor((timeGone / 86400) * 24)) Mod 24)
'because "mod" fails for doubles (should print "13.000...00")
Print timeGone Mod 24
```

Did not think that it makes a difference before appending the ".00000" but indeed it does:

Windows:
```
Building untitled3
Compiling:untitled3.bmx
flat assembler  version 1.69.14  (1137314 kilobytes memory)
3 passes, 917 bytes.
Linking:untitled3.exe
Executing:untitled3.exe
0
13.000000000000000

Process complete
```

Linux:
```
...
0
13.000000000000000
```

**Comment by woollybah** [comment]
It seems I shouldn't be adding "f" on the end...

**Comment by GWRon** [comment]
I do not know if your latest commit (removing the "f" for doubles) should fix the whole issue - so excuse if I write before you finished with your changes.

With the current commit the value is only correct for "doubleVar mod otherVar". 
It seems as if the Floor(double)-result contains ".00000" in all cases then - which corrupts the upcoming modulo operation.

**Comment by woollybah** [comment]
Are you having a problem with getting zero from your first calculation?

**Comment by woollybah** [comment]
I've no problems here with Floor. I can't see where you are having problems, other than the fact that ```(x * 24) mod 24``` will always equal zero?

**Comment by GWRon** [comment]
What do you mean?

Your commit only fixed the dirdct usage of 

DoubleWithoutDotzerozero mod value

If you use floor(calc) like in the second print statement, this appends .0000 to doubles and this results in the 0 instead of "13"

**Comment by woollybah** [comment]
Okay, please give me an example that isn't working.

**Comment by GWRon** [comment]
Do not skip the floor()...it floors the days (gets rid of hours 1-23 as they are on the same day).

my sample should return 13 in all cases. Regardless of .00000 or not .00000 .

I also still dont know what you meant with the value loosing "68" when converting from double to long

**Comment by woollybah** [comment]
The sample is working now - at least the second Print is. The first Print will print zero forever, regardless the language you code it in :rabbit:

**Comment by GWRon** [comment]
Doesnt the example above work (not work)?

With vanilla i get 13 as a result in my function. With NG i get 0.

**Comment by woollybah** [comment]
What function?
The example above, prints 0 and 13.

**Comment by GWRon** [comment]
Will check in 5 minutes when at the computer.

I use that formula in my worldtimeclass... and it does not return "0" there.

Will check that in some minutes

**Comment by GWRon** [comment]
Lol ...youare absolutely right.

Of course the time is "0" ... I dunno how I expected 13 ... Seems I need to catch some sleep.

Explanation: I took the timestamp of my game - which starts 00:00 but somehow I had the "13" in my mind. 
Man I tried to find out what could lead to a zero ... but math must be trusted.

Seems I mixed my "formula result"-expectation with the "modulo"-expectation (which indeed had to be "13".
